src/main/java/com/thealgorithms/others/TowerOfHanoi.java

```diff
@@ -1,29 +1,47 @@
 package com.thealgorithms.others;
 
-import java.util.Scanner;
+import java.util.List;
 
+/**
+ * The {@code TowerOfHanoi} class provides functionality to solve the Tower of
+ * Hanoi puzzle.
+ * It uses recursion to move discs between poles and prints the steps to solve
+ * the puzzle.
+ * The main function interacts with the user to get the number of discs and
+ * calls the recursive {@code shift} function to perform the moves.
+ * Wikipedia: https://en.wikipedia.org/wiki/Tower_of_Hanoi
+ */
```

Comment:
More detail description can be look like this:

```
/**
 * The {@code TowerOfHanoi} class provides a recursive solution to the Tower of Hanoi puzzle.
 * This puzzle involves moving a set of discs from one pole to another, following specific rules:
 * 1. Only one disc can be moved at a time.
 * 2. A disc can only be placed on top of a larger disc.
 * 3. All discs must start on one pole and end on another.
 * 
 * This implementation recursively calculates the steps required to solve the puzzle and stores them
 * in a provided list.
 * 
 * <p>
 * For more information about the Tower of Hanoi, see 
 * <a href="https://en.wikipedia.org/wiki/Tower_of_Hanoi">Tower of Hanoi on Wikipedia</a>.
 * </p>
 */
```

This description adds more details

Reply:
Done!